source/extensions/clusters/redis/redis_cluster_lb.cc

```diff
@@ -6,8 +6,18 @@ namespace Clusters {
 namespace Redis {
 
 bool ClusterSlot::operator==(const Envoy::Extensions::Clusters::Redis::ClusterSlot& rhs) const {
-  return start_ == rhs.start_ && end_ == rhs.end_ && primary_ == rhs.primary_ &&
-         replicas_ == rhs.replicas_;
+  if (start_ != rhs.start_ || end_ != rhs.end_ || *primary_ != *rhs.primary_ ||
+      replicas_.size() != rhs.replicas_.size()) {
+    return false;
+  }
+
+  for (auto const& replica : rhs.replicas_) {
+    if (replicas_.find(replica.first) == replicas_.end()) {
+      return false;
+    }
+  }
```

Comment:
suggestion: Currently this is O(n*log n). If the replicas are `std::map` then we can rely on how they are ordered like in your first commit to make this loop O(n).

Reply:
It's a good suggestion, thank you! I have fixed it and created a new commit.